test/v2/models/response/retrieveBalanceAndTransactions/CodingDetailsSpec.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.models.response.retrieveBalanceAndTransactions
+
+import play.api.libs.json.Json
+import support.UnitSpec
+import v2.fixtures.retrieveBalanceAndTransactions.CodingDetailsFixture._
+
+class CodingDetailsSpec extends UnitSpec {
+
+  "reads" should {
+    "return a valid model" when {
+      "valid JSON is supplied" in {
+        codingDetailsDownstreamDetailsJson
+          .as[CodingDetails] shouldBe codingDetailsObject
+      }
+    }
+  }
+
+  "writes" when {
+    "passed a valid model" should {
```

Comment:
can this be "passed a valid object" or something (again, not a model :slightly_smiling_face: )

Reply:
Done